searches/search_test.go

```diff
@@ -25,7 +32,7 @@ func TestBinarySearch(t *testing.T) {
 	for _, test := range searchTests {
 		actual := BinarySearch(test.data, test.key, 0, len(test.data)-1)
 		if actual != test.expected {
-			t.Errorf("test %s failed", test.name)
+			t.Errorf("test '%s' failed: input array '%v' with key '%d', expected '%d', get '%d'", test.name, test.data, test.key, test.expected, actual)
```

Comment:
If you add the error as a return type then it would be worth checking for the expected errors.